File: internal/model/policy.go
package model

import "time"

type Policy struct {
	ID                int64     `json:"id"`
	InstanceID        int64     `json:"instance_id"`
	Name              string    `json:"name"`
	Type              string    `json:"type"`
	TargetID          int64     `json:"target_id"`
	ScheduleType      string    `json:"schedule_type"`
	ScheduleValue     string    `json:"schedule_value"`
	Enabled           bool      `json:"enabled"`
	Compression       bool      `json:"compression"`
	Encryption        bool      `json:"encryption"`
	EncryptionKeyHash *string   `json:"-"`
	SplitEnabled      bool      `json:"split_enabled"`
	SplitSizeMB       *int      `json:"split_size_mb,omitempty"`
	RetentionType     string    `json:"retention_type"`
	RetentionValue    int       `json:"retention_value"`
	CreatedAt         time.Time `json:"created_at"`
	UpdatedAt         time.Time `json:"updated_at"`
}

type PolicyExecutionSummary struct {
	LastExecutionTime   *time.Time `json:"last_execution_time,omitempty"`
	LastExecutionStatus *string    `json:"last_execution_status,omitempty"`
	LatestBackupID      *int64     `json:"latest_backup_id,omitempty"`
}

File: web/src/types/policy.ts
export interface HookCommand {
  location: string
  command: string
}

export interface Policy {
  id: number
  instance_id: number
  name: string
  type: 'rolling' | 'cold'
  target_id: number
  schedule_type: 'interval' | 'cron'
  schedule_value: string
  bandwidth_limit_kb: number
  enabled: boolean
  compression: boolean
  encryption: boolean
  split_enabled: boolean
  split_size_mb?: number
  retry_enabled: boolean
  retry_max_retries: number
  retention_type: 'time' | 'count'
  retention_value: number
  pre_commands: HookCommand[]
  post_commands: HookCommand[]
  created_at: string
  updated_at: string
  last_execution_time?: string
  last_execution_status?: string
  latest_backup_id?: number
}

export interface CreatePolicyRequest {
  name: string
  type: 'rolling' | 'cold'
  target_id: number
  schedule_type: 'interval' | 'cron'
  schedule_value: string
  bandwidth_limit_kb: number
  enabled: boolean
  compression: boolean
  encryption: boolean
  encryption_key?: string
  split_enabled: boolean
  split_size_mb?: number
  retry_enabled: boolean
  retry_max_retries: number
  retention_type: 'time' | 'count'
  retention_value: number
  pre_commands: HookCommand[]
  post_commands: HookCommand[]
}

export interface UpdatePolicyRequest extends CreatePolicyRequest {}
